api/provider/aws/test_aws/service_create_test.go

```diff
@@ -0,0 +1,246 @@
+package test_aws
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/golang/mock/gomock"
+	provider "github.com/quintilesims/layer0/api/provider/aws"
+	"github.com/quintilesims/layer0/api/tag"
+	awsc "github.com/quintilesims/layer0/common/aws"
+	"github.com/quintilesims/layer0/common/config/mock_config"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestServiceCreate(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	defer provider.SetEntityIDGenerator("svc_id")()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	tags := models.Tags{
+		{
+			EntityID:   "dpl_id",
+			EntityType: "deploy",
+			Key:        "arn",
+			Value:      "dpl_arn",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// ServiceProvider.Create() makes AWS calls that need to be mocked out:
+	// - ELB.DescribeLoadBalancer (if req.LoadBalancerID != "")
+	// - ECS.DescribeTaskDefinition (if req.LoadBalancerID != "")
+	// - ECS.CreateService
+
+	// ELB.DescribeLoadBalancer
+	loadBalancerInput := &elb.DescribeLoadBalancersInput{}
+	loadBalancerInput.SetLoadBalancerNames([]*string{aws.String("l0-test-lb_id")})
+	loadBalancerInput.SetPageSize(1)
+
+	loadBalancerOuput := &elb.DescribeLoadBalancersOutput{}
```

Comment:
Not sure if this line is needed if you're leaving this empty. The return line would then look like `Return(&elb.DescribeLoadBalancersOutput{})`

Reply:
No, it's not needed - it was just a consistency-of-form thing that came out of skeletoning and that I never went back to remove.